pandas/core/sparse/frame.py

```diff
@@ -846,6 +846,13 @@ def applymap(self, func):
         return self.apply(lambda x: lmap(func, x))
 
 
+def _nan_to_np_nan(value):
+    """Normalize nan values to singleton np.NaN object so that when NaNs are
```

Comment:
Incorrectly in the sense that:
```py
>>> df = pd.DataFrame({np.nan: [1, 2]})
>>> df[np.nan]   # Arguably expectedly, nan matches nan
0    1
1    2
Name: nan, dtype: int64

>>> df = pd.DataFrame({np.nan: [1, 2], 2: [2, 3]}, columns=[np.nan, 2])
>>> df   # nan from dict didn't match the nan from ensured Float64index
  NaN    2.0
0  NaN     2
1  NaN     3
```
Nans are tricky because it generally holds `nan != nan`. But I guess more often than not,  this leads to confusion when nan is expected to be a single, catch-all bin.

How can I improve on the above singleton approach?